### PR TITLE
Refactor email API

### PR DESCRIPTION
**Pull request by ywk253100**

### CODE REVIEW

File: src/ui/api/email.go [hunk 1]
```diff
@@ -32,20 +30,20 @@ const (
...
 }
 
 // Prepare ...
 func (e *EmailAPI) Prepare() {
-	userID := e.ValidateUser()
-	isSysAdmin, err := dao.IsAdminRole(userID)
-	if err != nil {
-		log.Errorf("failed to check the role of user: %v", err)
-		e.CustomAbort(http.StatusInternalServerError, http.StatusText(http.StatusInternalServerError))
+	e.BaseController.Prepare()
```
Comment (wy65701436) [inline src/ui/api/email.go]: Is there any better way to have a general rule to check the authentication?  I see the duplicate code block in several APIs.

Comment (ywk253100) [inline src/ui/api/email.go]: Yes, we can add a filter to check the authentication. But it's complicate as not all API calls need authentication, we need a filter to do the authentication according to the url, method, etc. Maybe we can refactor this part after completing the current work.

Comment (wy65701436) [inline src/ui/api/email.go]: Can you please add an new issue for it? I can merge code firstly.

Comment (ywk253100) [inline src/ui/api/email.go]: OK. Thanks.

Comment (ywk253100) [inline src/ui/api/email.go]: Open issue #2426 to track this.

**Pull request merged**